go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -0,0 +1,221 @@
+// +build windows
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types/current"
+
+	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/config"
+)
+
+// noNameNetNS - This is received for the infra container, in this case we
+// have to create the network endpoint and attach it to the infra container
+// containerNetNS - This is received for all the other containers from
+// the same pod, do not create the network endpoint in thi case
+const (
+	noNameNetNS    = "none"
+	containerNetNS = "container:"
+)
+
+// More details about the above constants can be found in the following PR:
+// https://github.com/kubernetes/kubernetes/pull/51063
+
+/**
+ * When the HNS Endpoint is created, it asks for the HNS Id in order to
+ * attach it to the desired network. This function finds the HNS Id of the
+ * network based on the gatewayIP. If more than one suitable network it's found,
+ * return an error asking to give the HNS Network Id in config.
+ */
+func getHNSIdFromConfigOrByGatewayIP(gatewayIP string) (string, error) {
+	if config.CNI.WinHNSNetworkId != "" {
```

Comment:
small nit on this one, but probably can't remember it right :). WinHNSNetworkId is a GUID? If it is mind checking if it has a GUID format, if not please check the length to be a proper one.

Reply:
The validation will be done anyway during the HNS Endpoint creation, if it's an invalid GUID, it will say that it's an invalid GUID or the HNS Network with that GUID could not be found.

Reply:
True but the you can give a better error message here and the user knows to update the config :)